packages/core/src/helpers.ts

```diff
@@ -88,18 +130,74 @@ export function buildConversation(providerLog: ProviderLogDto) {
       } as ToolRequestContent
     })
 
-    if (message) {
-      message.content = (message.content as MessageContent[]).concat(content)
-      message.toolCalls = providerLog.toolCalls
-    } else {
-      message = {
-        role: MessageRole.assistant,
-        content: content,
-        toolCalls: providerLog.toolCalls,
+    content = content.concat(toolContents)
+  }
+
+  if (toolCallResponses.length > 0) {
+    const toolResponseContents = toolCallResponses.map((toolCallResponse) => {
+      return {
+        type: ContentType.toolResult,
+        toolCallId: toolCallResponse.id,
+        toolName: toolCallResponse.name,
+        result:
+          typeof toolCallResponse.result === 'string'
+            ? JSON.parse(toolCallResponse.result)
+            : toolCallResponse.result,
+        isError: toolCallResponse.isError || false,
```

Comment:
I don't like the fact that we're building response message with the types of old compiler. But I don't want to change this now.

We should do a PR killing all compiler types and start using prompt ASAP